go-controller/pkg/factory/factory.go

```diff
@@ -534,3 +577,51 @@ func (wf *WatchFactory) AddNodeHandler(handlerFuncs cache.ResourceEventHandler,
 func (wf *WatchFactory) RemoveNodeHandler(handler *Handler) error {
 	return wf.removeHandler(nodeType, handler)
 }
+
+// GetPod returns the pod spec given the namespace and pod name
+func (wf *WatchFactory) GetPod(namespace, name string) (*kapi.Pod, error) {
+	podLister := wf.informers[podType].lister.(*listers.PodLister)
```

Comment:
@abhat all these listers are just an interface themselves, so I don't think we need the `*` when casting. Interfaces are already pointers (more or less).

Reply:
I think there is still an inherent penalty when creating a "copy-by-value" of the `podLister` struct which implements the interface when NewPodLister() is called.

Reply:
@abhat there shouldn't be any copy-by-value here. The actual variable stored in the `lister` should be a pointer:

```
func newInformerLister(oType reflect.Type, sharedInformer cache.SharedIndexInformer) (listerInterface, error) {
	switch oType {
	case podType:
		return listers.NewPodLister(sharedInformer.GetIndexer()), nil
<snip>

// NewPodLister returns a new PodLister.
func NewPodLister(indexer cache.Indexer) PodLister {
	return &podLister{indexer: indexer}
}
```

newInformerLister returns an interface, but that interface will actually contain a pointer to a podLister object. So then we can just do this:

```
func (wf *WatchFactory) GetPod(namespace, name string) (*kapi.Pod, error) {
	podLister := wf.informers[podType].lister.(listers.PodLister)
	return podLister.Pods(namespace).Get(name)
}
```